test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,403 @@
+            log.info("Scope that is seen by readers {}", scope);
+            //start reading events
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader{}", i);
+                log.info("Creating reader with id {}", readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                        readerGroupName,
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+            readWriteTest(writerList, readerList, stopWriteFlag);
+
+            //close all the writers
+            writerList.forEach(writer -> writer.close());
+            //close all readers
+            readerList.forEach(reader -> reader.close());
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+            //seal the stream
+            CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(scope, STREAM_NAME);
+            log.info("Sealing stream {}", STREAM_NAME);
+            assertTrue(sealStreamStatus.get());
+            //delete the stream
+            CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+            log.info("Deleting stream {}", STREAM_NAME);
+            assertTrue(deleteStreamStatus.get());
+
+            //delete the scope
+            CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+            log.info("Deleting scope {}", scope);
+            assertTrue(deleteScopeStatus.get());
+            log.info("Test {} succeeds ", "MultiReaderWriterWithFailOver");
+        }
+
+    private void readWriteTest(final List<EventStreamWriter<Long>> writerList,
+                               List<EventStreamReader<Long>> readerList, final AtomicBoolean stopWriteFlag) throws InterruptedException, ExecutionException {
+        // start writing asynchronously
+        List<CompletableFuture<Void>> writerFutureList = new ArrayList<>();
+        writerList.forEach(writer -> {
+            CompletableFuture<Void> writerFuture = startWriting(eventData, writer, stopWriteFlag);
+            writerFutureList.add(writerFuture);
+        });
+
+        //start reading asynchronously
+        List<CompletableFuture<Void>> readerFutureList = new ArrayList<>();
+        readerList.forEach(reader -> {
+            CompletableFuture<Void> readerFuture = startReading(eventsReadFromPravega, eventData, eventReadCount, stopReadFlag, reader);
+            writerFutureList.add(readerFuture);
+        });
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        eventWriteCount1 = eventData;
+        eventReadCount1 =  eventReadCount;
+
+        //Scale down SSS instances to 2
```

Comment:
Instead of having the scaling happening here, which is a bit odd because it is not directly related to reading and writing, perhaps we should have a different method that we call say `performScaling()` and we call it immediately after `readWriteTest` in the main test case method. Another option is that we move all this code related to scaling to the main test case method. I'm fine either way.